### PR TITLE
Login procedure refactoring

### PR DESCRIPTION
**Pull request by YaroTs**
Errors handling modified

### CODE REVIEW

File: mobile/src/root.tsx [hunk 1]
```diff
@@ -21,11 +20,6 @@ export class Root extends Component<{}> {
         config.oauth.redirectUri);
 
       this.navigationService = new NavigationService();
-
-      const isIOS = Platform.OS === 'ios';
```
Comment (santee) [inline mobile/src/root.tsx]: does it still work on Android? I don't remember the code but from the looks of it, something else caused Android app to login, now you call it for both?

Comment (YaroTs) [reply mobile/src/root.tsx]: Yes, now the login procedure is started in App component for both platforms.

File: mobile/src/reducers/auth/auth.epics.ts [hunk 2]
```diff
@@ -54,16 +65,20 @@ export const listenerAuthStateEpic$ = (action$: ActionsObservable<any>, _: State
     dep.oauthProcess.authenticationState
         .pipe(
             handleHttpErrors(),
-            distinctUntilChanged<AuthenticationState>((x, y) => x.isAuthenticated === y.isAuthenticated),
+            distinctUntilChanged<AuthenticationState>((x, y) => ((x.isAuthenticated === y.isAuthenticated) && x.isAuthenticated)),
```
Comment (DesmanLead) [inline mobile/src/reducers/auth/auth.epics.ts]: Why is this required? Is x == next, y == previous (consider renaming)?

Comment (YaroTs) [reply mobile/src/reducers/auth/auth.epics.ts]: This is needed because if we can't authenticate several times (Authentication state is NotAuthenticated) , we anyway need to show error every time. Without this additional condition no error message will be shown.
